packages/flutter/lib/src/widgets/shortcuts.dart

```diff
@@ -274,60 +609,57 @@ class ShortcutManager extends ChangeNotifier with Diagnosticable {
   /// When the map is changed, listeners to this manager will be notified.
   ///
   /// The returned map should not be modified.
-  Map<LogicalKeySet, Intent> get shortcuts => _shortcuts;
-  Map<LogicalKeySet, Intent> _shortcuts;
-  set shortcuts(Map<LogicalKeySet, Intent> value) {
+  Map<ShortcutActivator, Intent> get shortcuts => _shortcuts;
+  Map<ShortcutActivator, Intent> _shortcuts = <ShortcutActivator, Intent>{};
+  set shortcuts(Map<ShortcutActivator, Intent> value) {
     assert(value != null);
-    if (!mapEquals<LogicalKeySet, Intent>(_shortcuts, value)) {
+    if (!mapEquals<ShortcutActivator, Intent>(_shortcuts, value)) {
       _shortcuts = value;
+      _indexedShortcutsCache = null;
       notifyListeners();
     }
   }
 
+  static Map<LogicalKeyboardKey?, List<_ActivatorIntentPair>> _indexShortcuts(Map<ShortcutActivator, Intent> source) {
+    final Map<LogicalKeyboardKey?, List<_ActivatorIntentPair>> result = <LogicalKeyboardKey?, List<_ActivatorIntentPair>>{};
+    source.forEach((ShortcutActivator activator, Intent intent) {
+      final Iterable<LogicalKeyboardKey?>? triggeringKeys = activator.triggers;
+      for (final LogicalKeyboardKey? trigger in triggeringKeys ?? <LogicalKeyboardKey?>[null]) {
```

Comment:
```suggestion
      for (final LogicalKeyboardKey? trigger in activator.triggers ?? <LogicalKeyboardKey?>[null]) {
```
Then you don't need the extra local variable, and I don't think it's any less readable.

Reply:
It's only because the compiler would complain "The type 'Object' used in the 'for' loop must implement Iterable." somehow. :(

Reply:
Weird. Maybe because it's nullable and null doesn't implement `Iterable`? Maybe make it non-nullable and move the `??` into the assignment:

```
      final Iterable<LogicalKeyboardKey?> triggeringKeys = activator.triggers ?? <LogicalKeyboardKey?>[null];
      for (final LogicalKeyboardKey? trigger in triggeringKeys) {
```

Reply:
Same error. I raised an issue to Dart https://github.com/dart-lang/language/issues/1572

Reply:
I've also added a comment here with reference to this issue.

Reply:
Actually, I just realized that the nullability here is an obsolete design and is no longer needed for now. This is no longer an issue.